uaclient/config.py

```diff
@@ -751,7 +751,104 @@ def _attached_status(self) -> Dict[str, Any]:
+
+        contract_information = get_contract_information(self, token)
+
+        contract_info = contract_information.get("contractInfo", {})
+        account_info = contract_information.get("accountInfo", {})
+
+        response = {
+            "version": version.get_version(features=self.features),
+            "contract": {
+                "id": contract_info.get("id", ""),
+                "name": contract_info.get("name", ""),
+                "created_at": contract_info.get("createdAt", ""),
+                "products": contract_info.get("products", []),
+            },
+            "account": {
+                "name": account_info.get("name", ""),
+                "id": account_info.get("id"),
+                "created_at": account_info.get("createdAt", ""),
+                "external_account_ids": account_info.get(
+                    "externalAccountIDs", []
+                ),
+            },
+            "services": [],
+            "execution_status": DEFAULT_STATUS.get("execution_status", ""),
+            "simulated": True,
+        }
+
+        if contract_info.get("effectiveTo"):
+            response["expires"] = contract_info.get("effectiveTo")
+        if contract_info.get("effectiveFrom"):
+            response["effective"] = contract_info.get("effectiveFrom")
+
+        resources = get_available_resources(self)
+        entitlements = contract_info.get("resourceEntitlements", [])
+
+        inapplicable_resources = {
+            resource["name"]: resource.get("description")
+            for resource in sorted(resources, key=lambda x: x["name"])
+            if not resource["available"]
+        }
+
+        for resource in sorted(resources, key=lambda x: x.get("name", "")):
+            entitlement_name = resource.get("name", "")
+            ent_cls = ENTITLEMENT_CLASS_BY_NAME.get(entitlement_name)
+            if ent_cls:
+                ent = ent_cls(self)
+                entitlement_information = self._get_entitlement_information(
+                    entitlements, entitlement_name
+                )
+                response["services"].append(
+                    {
+                        "name": ent.name,
+                        "description": ent.description,
+                        "entitled": entitlement_information["entitled"],
+                        "auto_enabled": entitlement_information[
+                            "auto_enabled"
+                        ],
+                        "available": "yes"
+                        if ent.name not in inapplicable_resources
+                        else "no",
+                    }
+                )
+
+        support = self._get_entitlement_information(entitlements, "support")
+        if support["entitled"]:
+            supportLevel = support["affordances"].get("supportLevel")
+            if supportLevel:
+                response["contract"]["tech_support_level"] = supportLevel
+
+        return response
+
+    def status(
```

Comment:
It seems that the only thing we are using from `status` is the beta service redaction part.
I think we should turn that beta part into a separate function and create function for the status simulation instead

Reply:
Indeed.
I just created here a similar flow to the other status calls we have, and there are (and there will be more) small parts that we need to use for them all, where some things can be simply separate.
I think the whole `status` flow needs to be refactored anyway - Taken out of the `UAConfig` class and taken care of.

My proposal here would be: can we merge this functionality first, and then work on a deeper refactoring of the status?
@orndorffgrant @blackboxsw thoughts?

Reply:
Oki so
I took a look on the refactoring work and I assume that your suggestion makes sense for this PR. When moving everything to a different module is is expected that those two pieces of functionality are isolated anyway. /me on it.